server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -177,10 +195,42 @@ fn generate(
         .map(|line| line.item_id)
         .collect();
 
-    let requisition_line_rows =
+    let requisition_lines =
         generate_requisition_lines(ctx, &ctx.store_id, &requisition, program_item_ids)?;
 
-    Ok((requisition, requisition_line_rows))
+    let program_indicators = program_indicators(
+        connection,
+        Pagination::all(),
+        None,
+        Some(ProgramIndicatorFilter::new().program_id(EqualFilter::equal_to(&program.id))),
+    )?;
+
+    let store = StoreRepository::new(connection)
+        .query_one(StoreFilter::new().name_id(EqualFilter::equal_to(&ctx.store_id)))?;
+
+    let should_generate_indicators: bool =
+        if let Some(supplying_store_id) = store.and_then(|s| s.name_row.supplying_store_id) {
+            supplying_store_id == other_party_id
```

Comment:
This is not checking the correct ids. the supplying_store_id is a store id but you are checking it against a name id.

Reply:
Hey @roxy-dao 

@Chris-Petty mentioned that some internal orders via programs will have an other party id which is not their typical supplying_store_id.

I wonder had you thought of that case? I am wondering if that check would still be valid.

Reply:
Had a convo with @Chris-Petty. Agree, that was something I've brought up before, but for now we will go with the supplying_store to generate indicators and a carry over issue should be created to allow stores to input indicators based on their facility level. #5844